model/tasks/SendAgsScoreTask.php

```diff
@@ -90,4 +106,38 @@ private function reportError(string $message): Report
 
         return Report::createError($message);
     }
+
+    private function retryTask(): void
+    {
+        if (!$this->isMaxRetryCountReached()) {
+            $this->increaseRetryCount();
+            $this->getQueueDispatcher()->createTask(new self, $this->params);
+            $this->logWarning('AGS score sending task has been retried');
+        }
+    }
+
+    public function increaseRetryCount(): void
+    {
+        $this->params[self::RETRY_COUNT]++;
```

Comment:
```suggestion
        @$this->params[self::RETRY_COUNT]++;
```

Just in case the retry count has not been provided.
Otherwise, this PR is a breaking change.

Reply:
I think adding default values to the expected params is easier to solve that and we also avoid spreading these conditions into the code base like in the method `isMaxRetryCountReached`.